### ISSUE TITLE
[Due for payment 2025-04-23] Android - Acct/ WS settings - Device buttons navigation bar missing borders

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53186]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.67.0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Open the app and log in
2. Create a workspace if there is no any
3. Navigate to the workspace settings
4. Enable More features and enable Workflows, Rules and Invoices options
5. Navigate to Profile, Invoices, Workflows. Rules and More features

## Expected Result:
Device buttons navigation bar has borders and visible

## Actual Result:
Device buttons navigation bar is missing borders and barely visible.
The same happens in account settings: Profile, Wallet, Security

## Workaround:
Unknown

## Platforms:

- [x] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/0ae6af77-339e-4d1a-86c5-e2061f235133

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021861690553730814393</li>
        <li>Upwork Job ID: 1861690553730814393</li>
        <li>Last Price Increase: 2024-11-27</li>
<li>Automatic offers: </li>
<ul>
<li>paultsimura | Reviewer | 105095159</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @johncschuster</details>

**Comment by mountiny** [comment]
@kirillzyusko @chrispader this is same as the https://github.com/Expensify/App/issues/53192 - maybe it can be fixed together

**Comment by mountiny** [comment]
@kirillzyusko can you please comment on this one too? it is a same root cause as these other too so lets just keep one main issue and then handle the safe area in one pr for all these issues
https://github.com/Expensify/App/issues/53192
https://github.com/Expensify/App/issues/53207

**Comment by kirillzyusko** [comment]
Commenting here for assignment 👋

**Comment by kirillzyusko** [comment]
@mountiny but this is not deploy blocker anymore, right? So we should focus on high priority/deploy blocker first and then come back to this issue, right?

**Comment by mountiny** [comment]
I will ask for a retest in staging

**Comment by mountiny** [comment]
bumped the thread

**Comment by mvtglobally** [comment]
Doesn't seem to be repro !

![Uploading 394867157...]

**Comment by mountiny** [comment]
@mvtglobally can you please specify the devices and android version - the videos are not loaded too

**Comment by mvtglobally** [comment]
Checking on devices @mountiny . No idea why these are no loading. Sent you in Slack.will try to reload in some time 
https://expensify.slack.com/archives/C9YU7BX5M/p1733941349787169?thread_ts=1733225032.044139&cid=C9YU7BX5M

**Comment by mvtglobally** [comment]
Mi11 Lite / Android 13 - Repro 
Samsung galaxy s10/ Android 12 - Not repro

**Comment by mountiny** [comment]
@kirillzyusko Seems like they can still reproduce 
https://github.com/user-attachments/assets/eee66f44-3d90-4298-b466-535a679a69fe

**Comment by kirillzyusko** [comment]
Okay, let me check it and prepare a fix 👀

**Comment by kirillzyusko** [comment]
@MelvinBot @chrispader is working on this 👀

**Comment by chrispader** [comment]
yes, i'm working on continuing the PR from @kirillzyusko.

**Comment by lakchote** [comment]
Not overdue, see https://expensify.slack.com/archives/C049HHMV9SM/p1737305732227189?thread_ts=1736238279.772929&cid=C049HHMV9SM

**Comment by johncschuster** [comment]
Bumping for Melvin. Not overdue!

**Comment by johncschuster** [comment]
Bumping for Melvin. Not overdue!

**Comment by johncschuster** [comment]
Bumping for Melvin. Not overdue!

**Comment by lakchote** [comment]
Still not overdue. Discussion is happening [here](https://expensify.slack.com/archives/C049HHMV9SM/p1737707357569159?thread_ts=1736238279.772929&cid=C049HHMV9SM).

**Comment by lakchote** [comment]
@chrispader is still working on this.

**Comment by johncschuster** [comment]
Not overdue. This is being worked on.

**Comment by mountiny** [comment]
@chrispader @kirillzyusko how is this one going?

**Comment by chrispader** [comment]
Gave an update on the progress here: https://github.com/Expensify/App/pull/55633#issuecomment-2663360315

I'm currently working on replacing the bottom safe area padding on all screens, and only apply it locally in scrollable content and on bottom-docked content in screens

**Comment by johncschuster** [comment]
Not overdue, Melv

**Comment by chrispader** [comment]
[EDITING]

Update on this: https://github.com/Expensify/App/pull/54285#issuecomment-2668222721



Additionally to the already discussed changes, there are some other changes that we should think about and fix in separate PRs. Once we disabled the bottom safe area handling completely in `ScreenWrapper`, we'll also have to make this consistent with e.g. modals and generic components like `SelectionList`, etc.

We might need to create follow-up-PRs for the following problems:

- **Modals:**
  - Modals that use the `BaseModal` component currently don't use the `ScreenWrapper` by default. Some do, but not all. We currently decide for each modal type individually about whether to apply safe area padding in `ModalStyleUtils.getModalStyles()`.
  - Ideally, we should not apply safe area padding in the root component of any modal, but rather only in the scrollable content and on bottom-docked content. Otherwise, scrollable content will not be able to go below the gesture bar and the user will always see a padding at the bottom, which does not look very good.
  - In a separate PR, we'll want to add a flag like `TEMPORARY_disableSafeAreaPaddingBottom` (or just `disableSafeAreaPaddingBottom`) with which we can selectively change this behavior for each modal indivudually.

**Comment by chrispader** [comment]
Please let me know if this makes sense to you or if you have any concerns! 🙌🏼

cc @mountiny @johncschuster @lakchote

**Comment by mountiny** [comment]
@chrispader I like the updated plan, we can also focus on first trying this out on handpicked modals/ components (that we feel like could be the most regression prone) to ensure no regressions are found in staging - or if yes, fix it without much disturbance

**Comment by lakchote** [comment]
I like your updated plan too @chrispader, and I'm with @mountiny on trying to do the updated logic first on App components that should be more prone to regressions.

Do you have an idea already on which components would you like to start with @chrispader?

**Comment by chrispader** [comment]
> Do you have an idea already on which components would you like to start with @chrispader?

Yes, i'd work on screens first, where we have to handle the keyboard, e.g. `ReportScreen` and all sorts of input screens in forms, etc.

Bottom safe area padding is most difficult to handle there. After that, most other screens should be very easy to migrate, since i implemented the safe area handling logic in each of the low-level components, namely:

 - `SelectionList`
 - `SectionList`
 - `ScrollView`
 - `ScreenWrapper`
 - `Modal`
 - (eventually more to come)

**Comment by johncschuster** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by johncschuster** [comment]
@lakchote can you take a look^?

**Comment by lakchote** [comment]
> [@lakchote](https://github.com/lakchote) can you take a look^?

It's not a blocker as it's only related to one platform and a minor UI issue, it's being fixed in https://github.com/Expensify/App/pull/58600

**Comment by johncschuster** [comment]
Awesome. Thank you!

**Comment by mountiny** [comment]
I think this is $250 to @hungvu193

**Comment by johncschuster** [comment]
## Payment Summary

Contributor: @kirillzyusko does not require payment (Contractor)
Contributor+: @hungvu193 owed $250 via NewDot

**Comment by johncschuster** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by johncschuster** [comment]
@hungvu193 can you complete the BZ Checklist above? Once complete, please go ahead and request payment vian NewDot. ([Summary here](https://github.com/Expensify/App/issues/53186#issuecomment-2748032707))

**Comment by hungvu193** [comment]
> [@hungvu193](https://github.com/hungvu193) can you complete the BZ Checklist above? Once complete, please go ahead and request payment via NewDot. ([Summary here](https://github.com/Expensify/App/issues/53186#issuecomment-2748032707))

Should I? I think there's another follow-up PR linked to this issue, and it's still open.

**Comment by johncschuster** [comment]
Oh! I didn't realize that. Thanks for the heads up, @hungvu193!

**Comment by garrettmknight** [comment]
$250 approved for @hungvu193

**Comment by hungvu193** [comment]
> [@hungvu193](https://github.com/hungvu193) can you link to the follow-up PR?

This one https://github.com/Expensify/App/pull/57281

**Comment by johncschuster** [comment]
Bumping for Melv. Payment has been issued for this. We're waiting on a follow-up PR to be completed in a different issue so the BZ Checklist can be completed here.

**Comment by mountiny** [comment]
@shubham1206agra is reviewing and @chrispader is ooo sick so awaiting progress on the PR for now

**Comment by mountiny** [comment]
I think this one should be completed now and ready for payment

**Comment by johncschuster** [comment]
This one was already paid out, [here](https://github.com/Expensify/App/issues/53186#issuecomment-2766257772). We're just waiting on the follow up PR so @hungvu193 can post the BZ Checklist.

**Comment by johncschuster** [comment]
## Payment Summary (to be issued after regression window has passed)

Contributor: @chrispader does not require payment (Contractor)
Contributor+: @hungvu193 does not require payment **(already paid, [here](https://github.com/Expensify/App/issues/53186#issuecomment-2766257772) ✅)**
Contributor+: @shubham1206agra owed $250 via NewDot

@shubham1206agra and @hungvu193 I see there are multiple PRs linked to this issue, and you both were the reviewers. Can you work together on the BZ Checklist above to come up with a holistic regression test?

**Comment by mountiny** [comment]
This was a large follow-up, so I think we should bump the price to @shubham1206agra to $375 - not more in my opinion, because there were two deploy blockers that @shubham1206agra was not around for to review, and so we ended up paying another C+ for help there. Unfortunately, the deploy blockers need to be resolved fast to unblock deploys.

@johncschuster If you dont have any concerns, could you please update the price to $375? thanks!

**Comment by johncschuster** [comment]
[Bumped in Slack](https://expensify.slack.com/archives/C02NK2DQWUX/p1745504980289459) for the BZ Checklist (particularly, the regression test steps) to be completed so we can close this out.

**Comment by johncschuster** [comment]
Still waiting on the BZ Checklist/regression test steps

**Comment by johncschuster** [comment]
Bumped in Slack^

**Comment by johncschuster** [comment]
Thank you! @shubham1206agra, please request payment via NewDot! (Summary [here](https://github.com/Expensify/App/issues/53186#issuecomment-2812877923))

**Comment by garrettmknight** [comment]
$375 approved for @shubham1206agra
